api/__tests__/index.test.js

```diff
@@ -0,0 +1,273 @@
+    });
+  });
+
+  describe('/:customerId/interests', () => {
+    it('Should respond to GET method', done => {
+      return request(app)
+        .get('/v1/customers/1/interests')
+        .set('Accept', 'application/json')
+        .expect(res => {
+          expect(res.body[0].name).toEqual("Traveling");
+        })
+        .expect(200, done);
+    });
+  });
+
+  describe('/:customerId/transactions', () => {
+    it('Should respond to GET method', done => {
+      return request(app)
+        .get('/v1/customers/1/transactions')
+        .set('Accept', 'application/json')
+        .expect(res => {
+          expect(res.body[0].customer_id).toEqual(1);
+        })
+        .expect(200, done);
+    });
+  });
+
+  describe('/:customerId/balances', () => {
+    it('Should respond to GET method', done => {
+      return request(app)
+        .get('/v1/customers/1/balances')
+        .set('Accept', 'application/json')
+        .expect(res => {
+          expect(res.body[0].name).toEqual('Air Miles');
+        })
+        .expect(200, done);
+    });
+  });
+});
+
+describe('Interests Endpoint', () => {
+  describe('/v1/interests', () => {
+    it('Should respond to GET method', done => {
+      request(app)
+        .get('/v1/interests')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(200, done);
+    });
+  });
+
+  describe('/v1/interests/:interestId', () => {
+    it('Should respond to GET method', done => {
+      request(app)
+        .get('/v1/interests/1')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(res => {
+          expect(res.body).toEqual({ id: 1, name: 'Traveling' });
+        })
+        .expect(200, done);
+    });
+
+    it("Should respond with 404 when requested item doesn't exist", done => {
+      request(app)
+        .get('/v1/interests/9999')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(404, done);
+    });
+  });
+});
+
+describe('Rewards Endpoint', () => {
+  describe('/v1/rewards', () => {
+    it('Should respond to GET method', done => {
+      request(app)
+        .get('/v1/rewards/')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(200, done);
+    });
+  });
+
+  describe('/v1/rewards/:rewardId', () => {
+    it('Should respond to GET method', done => {
+      request(app)
+        .get('/v1/rewards/1')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(res => {
+          expect(res.body).toEqual({ id: 1, name: 'Air Miles' });
+        })
+        .expect(200, done);
+    });
+
+    it("Should respond with 404 when requested item doesn't exist", done => {
+      request(app)
+        .get('/v1/rewards/9999')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(404, done);
+    });
+  });
+});
```

Comment:
Most of the previous comments apply for all the tests. The file might need a lot of rework.